test/jdk/java/lang/Class/getEnclosingClass/EnclosingClassTest.java

```diff
@@ -59,71 +48,153 @@
  * e o x x x
  */
 
-import java.util.List;
-import java.util.LinkedList;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
 import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
 import common.TestMe;
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.util.FileUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/*
+* @test
+* @bug 4992173 4992170
+* @library /test/lib
+* @modules jdk.compiler
+* @build jdk.test.lib.process.* EnclosingClassTest
+* @run testng/othervm EnclosingClassTest
+* @summary Check getEnclosingClass and other methods
+* @author Peter von der Ah\u00e9
+*/
+
 
 public class EnclosingClassTest {
-    static void info(Class<?> c, Class<?> encClass, String desc) {
-        if (!"".equals(desc))
+    private static final String SRC_DIR = System.getProperty("test.src");
+    private static final String ENCLOSING_CLASS_SRC = SRC_DIR + "/EnclosingClass.java";
+
+    @BeforeClass
+    public void createEnclosingClasses() throws Throwable {
+        Path enclosingPath = Paths.get(ENCLOSING_CLASS_SRC);
+        Path pkg1Dir = Paths.get(SRC_DIR + "/pkg1");
+        Path pkg2Dir = Paths.get(SRC_DIR+"/pkg1/pkg2");
+        Path pkg1File = Paths.get(SRC_DIR + "/pkg1/EnclosingClass.java");
+        Path pkg2File = Paths.get(SRC_DIR + "/pkg1/pkg2/EnclosingClass.java");
+
+        if(!Files.notExists(pkg1Dir)) {
+            FileUtils.deleteFileTreeWithRetry(pkg1Dir);
+        }
+        Files.createDirectories(pkg2Dir);
+        createAndWriteEnclosingClasses(enclosingPath, pkg1File, "pkg1");
+        createAndWriteEnclosingClasses(enclosingPath, pkg2File, "pkg1.pkg2");
+
+        String javacPath = JDKToolFinder.getJDKTool("javac");
+        OutputAnalyzer outputAnalyzer = ProcessTools.executeCommand(javacPath, "-d", System.getProperty("test.classes", "."),
+                SRC_DIR + "/EnclosingClass.java", SRC_DIR + "/pkg1/EnclosingClass.java", SRC_DIR + "/pkg1/pkg2/EnclosingClass.java");
```

Comment:
`File.separator` is different on Unix and Windows.   Either replace `/` with `File.separator`.   Or you can use `java.nio.file.Path` and call `Path::toString` to get the string representation.

See above suggestion of declaring `static final Path ENCLOSING_CLASS_SRC`.  So the above should use `ENCLOSING_CLASS_SRC` and `pkg2File` instead (calling toString)